src/main/java/controllers/ServletController.java

```diff
@@ -0,0 +1,124 @@
+    private static final WebCalculatorFactory FACTORY = new WebCalculatorFactory();
+    private static final Calculator CALCULATOR_SERVICE = FACTORY.createCalculator();
+    private static final Logger logger = LogManager.getLogger(ServletController.class);
+
+    public void getResult(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        HttpSession session = request.getSession();
+
+        String expression = (String) session.getAttribute(EXPRESSION);
+        Map<String, String> attributeValueMap = getAttributeValueMap(session);
+        PrintWriter writer = response.getWriter();
+        try {
+            int result = CALCULATOR_SERVICE.calculate(expression, attributeValueMap);
+            writer.print(result);
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (Exception e) {
+            logger.info(String.format(CALCULATE_EXCEPTION, expression));
+            response.sendError(HttpServletResponse.SC_CONFLICT, MISSING_EXPRESSION);
+        } finally {
+            writer.close();
+        }
+    }
+
+    public void deleteData(HttpServletRequest request, HttpServletResponse response) {
+        HttpSession session = request.getSession();
+        String attributeName = getItemNameFromURI(request);
+        session.removeAttribute(attributeName);
+        response.setStatus(HttpServletResponse.SC_NO_CONTENT);
+    }
+
+    public void putData(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        HttpSession session = request.getSession();
+        String attribute = getItemNameFromURI(request);
+        String value = request.getReader().readLine();
+
+        if (attribute.equalsIgnoreCase(EXPRESSION)) {
+            if (isGoodFormatExpression(value)) {
+                addData(response, session, value, attribute);
+            } else {
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST, WRONG_EXPRESSION);
+            }
+        } else {
+            if (!isParameterHasOverLimitValue(value)) {
+                addData(response, session, value, attribute);
+            } else {
+                response.sendError(HttpServletResponse.SC_FORBIDDEN, OVER_RANGE);
+            }
+        }
+
+        String requestURI = request.getRequestURI();
+        response.addHeader(LOCATION, requestURI);
+    }
+
+    private Map<String, String> getAttributeValueMap(HttpSession session) {
+        Enumeration<String> attributeNames = session.getAttributeNames();
+        Map<String, String> attributeValueMap = new ConcurrentHashMap<>();
+
+        while (attributeNames.hasMoreElements()) {
+            String attributeName = attributeNames.nextElement();
+            attributeValueMap.put(attributeName, (String) session.getAttribute(attributeName));
+        }
+
+        return attributeValueMap;
+    }
+
+    private String getItemNameFromURI(HttpServletRequest request) {
+        return request.getRequestURI().substring(URL_SUBSTRING);
```

Comment:
Request url can not to store itemName at all. How will the system behave in this case?